tests/lib/rules/named-functions-in-promises.js

```diff
@@ -39,6 +39,30 @@ eslintTester.run('named-functions-in-promises', rule, {
     }, {
       code: 'user.save().finally(_finallyDo);',
       parserOptions: { ecmaVersion: 6 },
+    }, {
+      code: 'user.save().then(() => this._reloadUser(user));',
+      parserOptions: { ecmaVersion: 6 },
+      options: [{
+        allowConciseArrow: true,
+      }],
+    }, {
+      code: 'user.save().catch(err => this._handleError(err));',
+      parserOptions: { ecmaVersion: 6 },
+      options: [{
+        allowConciseArrow: true,
+      }],
+    }, {
+      code: 'user.save().finally(() => this._finallyDo());',
+      parserOptions: { ecmaVersion: 6 },
+      options: [{
+        allowConciseArrow: true,
+      }],
+    }, {
+      code: 'user.save().then(() => user.reload());',
```

Comment:
I was just speaking with @bardzusny and we agreed that in this PR it's ok to allow using `xxx.method()` to keep it consistent with current regular function expressions that are allowed. We do however think that it might be worth considering in next PR to add additional optional setting called for example `requireThisBinding`. This setting in practice would force you to use `this.method.bind(this)` or `() => this.method()`. I'll create separate issue with this proposition though.